### PR TITLE
Disabled PCA inputs for deepTau v2.

### PR DESCRIPTION
**Pull request by swozniewski**

### CODE REVIEW

**Comment by kandrosov** [comment]
> Should that be a float, i.e. 0.f or similar?

@steggema in case of 0 it is equivalent, i.e. integer 0 will be implicitly converted to float 0 at the compile-time without any side effects.

File: RecoTauTag/RecoTau/plugins/DeepTauId.cc [hunk 1]
```diff
@@ -907,9 +907,9 @@ class DeepTauId : public deep_tau::DeepTauBase {
         get(dnn::neutralIsoPtSumdR03_over_dR05) = getValue(tau.tauID("neutralIsoPtSumdR03") / tau.tauID("neutralIsoPtSum"));
         get(dnn::photonPtSumOutsideSignalCone) = getValueNorm(tau.tauID("photonPtSumOutsideSignalConedR03"), 1.731f, 6.846f);
         get(dnn::puCorrPtSum) = getValueNorm(tau.tauID("puCorrPtSum"), 22.38f, 16.34f);
-        get(dnn::tau_dxy_pca_x) = getValueNorm(tau.dxy_PCA().x(), -0.0241f, 0.0074f);
-        get(dnn::tau_dxy_pca_y) = getValueNorm(tau.dxy_PCA().y(),0.0675f, 0.0128f);
-        get(dnn::tau_dxy_pca_z) = getValueNorm(tau.dxy_PCA().z(), 0.7973f, 3.456f);
+        get(dnn::tau_dxy_pca_x) = 0; // getValueNorm(tau.dxy_PCA().x(), -0.0241f, 0.0074f);
```
Comment (mbluj) [inline RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: Hi,
I will add a comment why zero is replacing the PCA point and why the original code is kept as set of commented lines. If it is not done I suppose such a comment from RECO. BTW, is it mandatory to keep original code knowing that usage of absolute PCA coordinates is meaningless?

Comment (kandrosov) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: > BTW, is it mandatory to keep original code knowing that usage of absolute PCA coordinates is meaningless?

The commented part can be removed if you prefer. I kept it to keep track of what input is expected by the NN (i.e. which variable range was used during the training).

Comment (mbluj) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: So, let's keep it. But, please add a line or two explaining the reasons of zeroing and keeping commented code. Not big deal but useful, no?

Comment (swozniewski) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: Well, in theory the neural net could still use PCA if we had fixed samples with comparable contents in data and MC (won't happen very soon of course...). So it's maybe not bad to keep the commented part to see what the net expects as Konstantin stated.
@mbluj please go ahead with adding a comment to the PRs (both for master and backport are set up). I guess it's a good idea to add it in advance. Will you also notify XPOG?

Comment (swozniewski) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: Crossing messages... Do I understand correctly that you would add the comment in the code not the PR description? Would indeed be good to stay aware of the reason

Comment (kandrosov) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: Yes, sure. I committed a comment that explains the reason of zeroing the PCA inputs: https://github.com/kandrosov/cmssw/commit/6c4e57e061c4aae9f48dbae3f987dc130a4f1a44 Please, let me know if it is ok.

Comment (mbluj) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: Yes, I mean to have comment also in code, I can add it if not done by Konstantin. I can notify XPOG, but I think it is better to to so when PR to 10_2_X is created. Do you want me create the set of PRs?
Concerning adding back the non-PV-relative PCA point to the NN - I am skeptical as it is not granted (or even expected) that position of BS is same in data and MC.

Comment (swozniewski) [reply RecoTauTag/RecoTau/plugins/DeepTauId.cc:910]: The PRs already exist. I can add Konstantin's comment maybe also adding the note `Kept as comment to document what NN expects` and applying the code formatting request by jenkins. It wants two spaces in front of the comment

**Comment by mbluj** [comment]
On Tue, Aug 27, 2019 at 11:22 AM swozniewski <notifications@github.com>
wrote:

> *@swozniewski* commented on this pull request.
> ------------------------------
>
> In RecoTauTag/RecoTau/plugins/DeepTauId.cc
> <https://github.com/cms-tau-pog/cmssw/pull/131#discussion_r317978298>:
>
> > @@ -907,9 +907,9 @@ class DeepTauId : public deep_tau::DeepTauBase {
>          get(dnn::neutralIsoPtSumdR03_over_dR05) = getValue(tau.tauID("neutralIsoPtSumdR03") / tau.tauID("neutralIsoPtSum"));
>          get(dnn::photonPtSumOutsideSignalCone) = getValueNorm(tau.tauID("photonPtSumOutsideSignalConedR03"), 1.731f, 6.846f);
>          get(dnn::puCorrPtSum) = getValueNorm(tau.tauID("puCorrPtSum"), 22.38f, 16.34f);
> -        get(dnn::tau_dxy_pca_x) = getValueNorm(tau.dxy_PCA().x(), -0.0241f, 0.0074f);
> -        get(dnn::tau_dxy_pca_y) = getValueNorm(tau.dxy_PCA().y(),0.0675f, 0.0128f);
> -        get(dnn::tau_dxy_pca_z) = getValueNorm(tau.dxy_PCA().z(), 0.7973f, 3.456f);
> +        get(dnn::tau_dxy_pca_x) = 0; // getValueNorm(tau.dxy_PCA().x(), -0.0241f, 0.0074f);
>
> The PRs already exist. I can add Konstantin's comment maybe also adding
> the note Kept as comment to document what NN expects and applying the
> code formatting request by jenkins. It wants two spaces in front of the
> comment
>
OK, it sounds good, please proceed. Thanks! I'll notify XPOG by adding a
comment in a PR which adds DeepTau to official NanoAOD.

**Comment by swozniewski** [comment]
done

**Pull request merged**